### PR TITLE
Fix cfn permissions

### PR DESCRIPTION
**Pull request by iakov-aws**
*Issue #, if available:*

*Description of changes:*
This fixes CFN permissions, making them more strict. Also fixes some test issues as well  
- make the kms permissions optional and configurable
- where possible use WorkGroup and Database permissions
- limit permissions on user ListUsers and DescribeUser

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: cfn-templates/cid-cfn.yml [hunk 1]
```diff
@@ -202,6 +209,17 @@ Conditions:
         - !Equals [ !Ref DeployCostIntelligenceDashboard, "yes" ]
         - !Equals [ !Ref DeployKPIDashboard, "yes" ]
   NeedRefreshDatasets:  !Not [ !Equals  [ !Ref QuickSightDataSetRefreshSchedule, ""] ]
+  NeedDataBucketsKms: !Equals [ !Ref DataBuketsKmsKeyArns, "" ]
+  NeedDataBucketsKmsAndNeedCURTable:
+    Fn::And:
```
Comment (tbuatois) [inline cfn-templates/cid-cfn.yml:214]: You can simplify all conditions by referencing them. 
Use !Condition NeedDataBucketsKms to reference boolean value of NeedDataBucketsKms condition
See https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/conditions-section-structure.html

Comment (iakov-aws) [reply cfn-templates/cid-cfn.yml:214]: Cool! This will really simplify this code! Thanks!

Comment (iakov-aws) [reply cfn-templates/cid-cfn.yml:208]: done https://github.com/aws-samples/aws-cudos-framework-deployment/pull/436/commits/005e9c9c4f0532214c9de5b0bdecf61fa7dc4085

File: cfn-templates/cid-cfn.yml [hunk 2]
```diff
@@ -68,11 +69,13 @@ Metadata:
       Suffix:
         default: "Suffix - Please do not change"
       QuickSightDataSetRefreshSchedule:
-        default: "QuickSight DataSet Refresh Schedule"
+        default: "QuickSight DataSet Refresh Schedule. Must be a valid cron or empty. If empty refresh will be disabled."
```
Comment (yprikhodko) [inline cfn-templates/cid-cfn.yml:72]: Maybe default to daily? It's the most common use case. Or another option is we could provide cron example for daily here

**Review by yprikhodko — APPROVED** [review]

File: cfn-templates/cid-cfn.yml (hunk 2 above)
Comment (iakov-aws) [reply cfn-templates/cid-cfn.yml:72]: default is 
`    Default: cron(0 4 * * ? *)`
https://github.com/aws-samples/aws-cudos-framework-deployment/blob/005e9c9c4f0532214c9de5b0bdecf61fa7dc4085/cfn-templates/cid-cfn.yml#L104

**Pull request merged**